### PR TITLE
bybit: migrate spot margin endpoints to v5 and align test/fixture coverage

### LINKED ISSUE #2121: Bybit: Fix live test failures due to endpoints being deprecated
```
--- FAIL: TestWSHandleAuthenticatedData (0.14s)
    bybit_test.go:3058:
                Error Trace:    C:/Users/AdrianGallagher/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/bybit/bybit_test.go:3058
                Error:          "0xc000daae70" should have 6 item(s), but has 5
                Test:           TestWSHandleAuthenticatedData
                Messages:       Should see correct number of messages
--- FAIL: TestGetBorrowableCoinInfo (0.30s)
    bybit_test.go:2560: Bybit unsuccessful HTTP status code: 404 raw response:
--- FAIL: TestGetVIPMarginData (0.35s)
    bybit_test.go:2552: Bybit unsuccessful HTTP status code: 404 raw response:
--- FAIL: TestGetMarginCoinInfo (0.58s)
    bybit_test.go:2544: Bybit unsuccessful HTTP status code: 404 raw response:
```

### PR DESCRIPTION
**Pull request by thrasher-**
## Summary
Closes #2121.

Migrates Bybit spot-margin calls to current v5 routes and aligns parsing/tests with live response shapes.

## Changes
- Move to:
  - `spot-margin-trade/data`
  - `spot-margin-trade/collateral`
  - `spot-margin-trade/max-borrowable` (auth v5)
- Parse collateral response via `collateralRatioList` (tiered shape).
- Update fixtures and tests for migrated query/response shapes.
- Stabilise Bybit tests in both mock and `mock_test_off` modes (including historical volatility + WS auth handling).
- Keep direct `retExtInfo` parsing coverage and misc assertion-style compliance.

## Validation
- `go test ./exchanges/bybit -count=1`
- `go test -tags mock_test_off ./exchanges/bybit -count=1`
- `make lint`
- `codespell`
- local misc-check simulation (workflow parity)

All green.

### CODE REVIEW

File: exchanges/bybit/bybit.go [hunk 1]
```diff
@@ -2097,32 +2098,59 @@ func (e *Exchange) GetVIPMarginData(ctx context.Context, vipLevel, ccy string) (
...
+	for i := range newResp.List {
+		conversionRate := types.Number(0)
+		if len(newResp.List[i].CollateralRatioList) > 0 {
+			conversionRate = newResp.List[i].CollateralRatioList[0].CollateralRatio
+		}
+		resp = append(resp, MarginCoinInfo{
+			Coin:           newResp.List[i].Coin,
+			ConversionRate: conversionRate,
+		})
+	}
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bybit/bybit.go]: Jarvis: Applied, thanks. Switched this to the idiomatic range-value form for readability (`for _, item := range newResp.List`) in b17afb47.

File: exchanges/bybit/bybit_types.go [hunk 2]
```diff
@@ -102,15 +102,38 @@ type InstrumentInfo struct {
...
 
+type retExtInfo struct {
+	List []ErrorMessage `json:"list"`
+}
+
+func (r *retExtInfo) UnmarshalJSON(data []byte) error {
+	if string(data) == "\"\"" || string(data) == "null" {
+		*r = retExtInfo{}
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bybit/bybit_types.go:113]: Jarvis: Reviewed with Codex + Claude and keeping this guard as-is for now. While simplifiable, the explicit `""`/`null` fast-path keeps Bybit edge-case intent obvious and aligned with existing `retExtInfo` tests.

File: exchanges/bybit/bybit.go [hunk 3]
```diff
@@ -2097,32 +2098,59 @@ func (e *Exchange) GetVIPMarginData(ctx context.Context, vipLevel, ccy string) (
...
 	}
+	params := url.Values{}
+	params.Set("currency", coin.String())
+
 	resp := struct {
-		List []BorrowableCoinInfo `json:"list"`
+		List []MaxBorrowableAmount `json:"list"`
 	}{}
-	return resp.List, e.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues("spot-cross-margin-trade/borrow-token", params), defaultEPL, &resp)
+	if err := e.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/spot-margin-trade/max-borrowable", params, nil, &resp, getSpotCrossMarginTradeLoanInfoEPL); err != nil {
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bybit/bybit.go]: Jarvis: Applied, thanks. Added a dedicated endpoint-limit constant (`getSpotMarginTradeMaxBorrowableEPL`) in `ratelimit.go` and switched `GetMaxBorrowableAmount` to use it in d1c11080.

File: exchanges/bybit/bybit_types.go [hunk 4]
```diff
@@ -102,15 +102,38 @@ type InstrumentInfo struct {
...
+	type alias retExtInfo
+	var parsed alias
+	if err := json.Unmarshal(data, &parsed); err != nil {
+		// Some Bybit responses can return retExtInfo as a raw string. Treat this
+		// shape as empty metadata so callers can still process successful payloads.
+		var ignored string
+		if errString := json.Unmarshal(data, &ignored); errString == nil {
+			*r = retExtInfo{}
+			return nil
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bybit/bybit_types.go:122]: Jarvis: Good catch — fixed in d1c11080. `SendAuthHTTPRequestV5` now treats any non-zero `retCode` as an auth failure (including the empty `retMsg` + string `retExtInfo` shape), and I added focused regression coverage in `bybit_auth_test.go` for this path.

File: exchanges/bybit/bybit_test.go [hunk 5]
```diff
@@ -727,17 +728,16 @@ func TestGetOpenInterestData(t *testing.T) {
 
 func TestGetHistoricalVolatility(t *testing.T) {
 	t.Parallel()
-	start := time.Now().Add(-time.Hour * 30 * 24)
-	end := time.Now()
+	var start, end time.Time
+	baseCoin := "BTC"
```
Comment (shazbert) [inline exchanges/bybit/bybit_test.go]: suggestion: just for standardisation we should be using currency.Code can also update guidelines that this type is preferred instead of string type for currency codes even the response field types.

File: exchanges/bybit/bybit_types.go [hunk 6]
```diff
@@ -102,15 +102,38 @@ type InstrumentInfo struct {
 	SettleCoin         currency.Code `json:"settleCoin"`
 }
 
+type retExtInfo struct {
+	List []ErrorMessage `json:"list"`
+}
+
+func (r *retExtInfo) UnmarshalJSON(data []byte) error {
+	if string(data) == "\"\"" || string(data) == "null" {
+		*r = retExtInfo{}
```
Comment (shazbert) [inline exchanges/bybit/bybit_types.go:111]: You can break recursion directly and not have to allocated again:

```diff
diff --git a/exchanges/bybit/bybit_types.go b/exchanges/bybit/bybit_types.go
index e428c6351..be306ac78 100644
--- a/exchanges/bybit/bybit_types.go
+++ b/exchanges/bybit/bybit_types.go
@@ -108,22 +108,17 @@ type retExtInfo struct {

 func (r *retExtInfo) UnmarshalJSON(data []byte) error {
        if string(data) == "\"\"" || string(data) == "null" {
-               *r = retExtInfo{}
                return nil
        }
-       type alias retExtInfo
-       var parsed alias
-       if err := json.Unmarshal(data, &parsed); err != nil {
+       type alias *retExtInfo
+       if err := json.Unmarshal(data, (alias)(r)); err != nil {
                // Some Bybit responses can return retExtInfo as a raw string. Treat this
                // shape as empty metadata so callers can still process successful payloads.
-               var ignored string
-               if errString := json.Unmarshal(data, &ignored); errString == nil {
-                       *r = retExtInfo{}
+               if errString := json.Unmarshal(data, new(string)); errString == nil {
                        return nil
                }
                return err
        }
-       *r = retExtInfo(parsed)
        return nil
 }
```

**Review by shazbert — CHANGES_REQUESTED** [review]
Not much from me. Thanks!

File: exchanges/bybit/bybit_test.go (hunk 5 above)
Comment (thrasher-) [reply exchanges/bybit/bybit_test.go]: Jarvis: Implemented, good suggestion — updated `GetHistoricalVolatility` to use `currency.Code` for `baseCoin`. Commit: `5ae28bd5`

File: exchanges/bybit/bybit_types.go (hunk 6 above)
Comment (thrasher-) [reply exchanges/bybit/bybit_types.go:111]: Jarvis: Thanks — I looked at this, and I’m keeping the current shape intentionally. The proposed change is a micro-optimisation (tiny allocation/copy difference) with negligible impact here, while the current version is more explicit/defensive (notably around reset behaviour for `null`/`""`/string fallback). Happy to revisit if we do a broader perf-focused cleanup with benchmarks.
